tests/backend/Functions.Tests/Mock/MockCollector.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
```

Comment:
I believe we are already using NSubstitute for mocking within the project. Instead of creating manual mocks like this, you can just use that. ie `var collector = Substitute.For<ICollector<T>>()`